web/app/controllers/login.js

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import Ember from 'ember';
+
+
+export default Ember.Controller.extend({
+  session: Ember.inject.service(),
+  error: '',
+  showError: false,
+  schedulerUrl: null,
+  cluster: null,
+  actions: {
+    login() {
+      let {username, password} = this.getProperties('username', 'password');
+      username = this.getValidStringValue(username)
+      password = this.getValidStringValue(password)
+      if (!username.length || !password.length) {
+        alert("Username or Password cannot be empty.")
+      } else if (this.cluster == null || this.schedulerUrl == null){
+        /*if cluster or the schedulerUrl is empty this means that
+           login route is not transitioned from the job page */
+        alert("Cannot process your request!! Visit the Job Page again and retry")
+      } else {
+          //session service will return an Ajax promise as a response of the POST call to /login API
+          this.get("session").login(username, password, this.schedulerUrl).then((response) => {
+            if (response.hasOwnProperty("status") && response.status === "success") {
+            //Setting same TTL for the cookie as the TTL of Azkaban session.id i.e. 10 hours
+              var inTenHours = new Date(new Date().getTime() + 10 * 60 * 60 * 1000);
+              const cookieName = "elephant." + this.cluster + ".session.id";
```

Comment:
Is the cluster a unique way to identify? Not sure if you would want to suffix session.id (which is in quotes) which cookieName as well.